src/App.tsx

```diff
@@ -14,37 +14,85 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  None = '',
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+}
 
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = [...goodsFromServer];
+
+  if (sortType === SortType.Alphabetical) {
+    visibleGoods.sort();
+  } else if (sortType === SortType.Length) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
```

Comment:
Please, prefer switch/case over multiple if/else

It may not be critical on two possible SortTypes, but still readability of the code suffers.